crates/matrix-sdk-indexeddb/src/crypto_store.rs

```diff
@@ -144,11 +144,14 @@ impl IndexeddbCryptoStore {
         let name = format!("{prefix:0}::matrix-sdk-crypto");
 
         // Open my_db v1
-        let mut db_req: OpenDbRequest = IdbDatabase::open_f64(&name, 2.0)?;
+        let mut db_req: OpenDbRequest = IdbDatabase::open_u32(&name, 2)?;
         db_req.set_on_upgrade_needed(Some(|evt: &IdbVersionChangeEvent| -> Result<(), JsValue> {
-            let old_version = evt.old_version();
+            // Even if the web-sys bindings expose the version as a f64, the IndexedDB API
+            // works with an unsigned integer.
+            // See <https://github.com/rustwasm/wasm-bindgen/issues/1149>
+            let old_version = evt.old_version() as u32;
```

Comment:
Too bad [`FloatToInt` is still unstable](https://doc.rust-lang.org/std/convert/trait.FloatToInt.html) at the time of writing. 

I would be a little more careful here than a cast.

What about:

```rust
let old_version = u32::try_from(evt.old_version().trunc() as i64).expect("blah blah");
```

There is still one caveat though. We should handle `evt.old_version().fract()`. We had the version 1.1 after all.

With this current patch:

* If client has version lower than 1, it will receive all updates.
* If client has version 1.0, it will receive updates for 1.1 and 2.
* If client has version 1.1, it will receive updates for 2.
* If client has version 2, it will receive zero update so far.

I guess it's correct with the current modifications. So, maybe use `trunc()` instead of using a cast `as u32`, but apart from that, it seems fine to not handle the case where `fract()` is not 0.0.

Reply:
It's not possible for `fract()` to not be `0.0`. The indexeddb API only works with integers so version `1.1` was actually always stored as version `1` just as version `1.0`. I'm pretty sure it actually never triggered a migration. That's why I'm proposing this change, so a confusion like this doesn't happen again in the future.

The only reason that the bindings use a `f64` by default is because the value can be bigger than a `u32`, but not as big as a `u64`.

Reply:
They should just use `js_int` x)

Reply:
Ah OK got it. Then it's good to me :-).